opensearch/src/main/java/org/opensearch/sql/opensearch/response/OpenSearchResponse.java

```diff
@@ -91,7 +94,21 @@ public Iterator<ExprValue> iterator() {
       }).iterator();
     } else {
       return Arrays.stream(hits.getHits())
-          .map(hit -> (ExprValue) exprValueFactory.construct(hit.getSourceAsString())).iterator();
+          .map(hit -> {
+            ExprValue docData = exprValueFactory.construct(hit.getSourceAsString());
+            if (hit.getHighlightFields().isEmpty()) {
+              return docData;
+            } else {
```

Comment:
else not needed. 

If we go into that branch, we return, so the else statement just adds a needless indent and statement.